### PR TITLE
Add an option for sharing a link to the comments.

### PR DESCRIPTION
**Pull request by jolteony**
Create a constant in common/Constants.java called SHARE_COMMENTS for use in CommentsListActivity.java.

Create a menu option when holding on a post when in the comments page to share a link to the comments. (As opposed to a link to the item that was submitted.) Shares the title of the post and the link to the comments.

### CODE REVIEW

**Comment by jolteony** [comment]
I tested this with @billhubgit 's gradle build that he gave me on reddit at billhubgit/diode-working. I don't actually know if it will run on the standard setup with ant, so please try it and see if it works, although I only changed two files within the program so I don't think it should have messed anything up.

Also, closes #54

**Comment by zagaberoo** [comment]
I'm really glad you were able to get a build system working!  These changes won't cause any trouble for the  Ant build system.

File: src/in/shick/diode/comments/CommentsListActivity.java [hunk 1]
```diff
@@ -1473,6 +1473,7 @@ public void onCreateContextMenu(ContextMenu menu, View v, ContextMenuInfo menuIn
 
         if (rowId == 0) {
             menu.add(0, Constants.SHARE_CONTEXT_ITEM, Menu.NONE, "Share");
+            menu.add(0, Constants.SHARE_COMMENTS, Menu.NONE, "Share Comments");
```
Comment (zagaberoo) [inline src/in/shick/diode/comments/CommentsListActivity.java]: I know the codebase is horribly inconsistent about this, but it's better to put this string into strings.xml (https://github.com/zagaberoo/diode/blob/master/res/values/strings.xml) and just reference it's R.string constant here.

File: src/in/shick/diode/comments/CommentsListActivity.java [hunk 2]
```diff
@@ -1669,6 +1670,22 @@ public boolean onContextItemSelected(MenuItem item) {
...
+        case Constants.SHARE_COMMENTS:
+            Intent intent2 = new Intent();
+            intent2.setAction(Intent.ACTION_SEND);
+            intent2.setType("text/plain");
+
+            intent2.putExtra(Intent.EXTRA_SUBJECT, getOpThingInfo().getTitle());
+            intent2.putExtra(Intent.EXTRA_TEXT, "https://reddit.com/r/" + mSubreddit + "/" + mThreadId );
+
+            try {
+                startActivity(Intent.createChooser(intent2, "Share Comments"));
```
Comment (zagaberoo) [inline src/in/shick/diode/comments/CommentsListActivity.java]: Same here as line 1476

File: src/in/shick/diode/comments/CommentsListActivity.java [hunk 3]
```diff
@@ -1669,6 +1670,22 @@ public boolean onContextItemSelected(MenuItem item) {
...
             getNewDownloadCommentsTask().execute(Constants.DEFAULT_COMMENT_DOWNLOAD_LIMIT);
             return true;
+            
+        case Constants.SHARE_COMMENTS:
+            Intent intent2 = new Intent();
+            intent2.setAction(Intent.ACTION_SEND);
+            intent2.setType("text/plain");
+
+            intent2.putExtra(Intent.EXTRA_SUBJECT, getOpThingInfo().getTitle());
+            intent2.putExtra(Intent.EXTRA_TEXT, "https://reddit.com/r/" + mSubreddit + "/" + mThreadId );
```
Comment (zagaberoo) [inline src/in/shick/diode/comments/CommentsListActivity.java]: The base url constant (https://github.com/zagaberoo/diode/blob/master/src/in/shick/diode/common/Constants.java#L358) would be a good fit in this line.

**Comment by jolteony** [comment]
I updated the files.

**Comment by zagaberoo** [comment]
Great work; thanks for contributing.

**Comment by jolteony** [comment]
:D My first pull request. Thanks!

**Comment by zagaberoo** [comment]
:)

**Pull request merged**